vllm/model_executor/models/mixtral_quant.py

```diff
@@ -48,7 +48,7 @@
 from vllm.model_executor.model_loader.weight_utils import default_weight_loader
 from vllm.model_executor.sampling_metadata import SamplingMetadata
 from vllm.sequence import IntermediateTensors, SamplerOutput
-
+from vllm.model_executor.layers.fused_moe import FusedMoE
```

Comment:
Sure. That makes a lot of sense. I'll try to allocate some time in recent 1-2 weeks to rebase and update.

Reply:
Hey @mgoin since I'm also using the compressed tensor format form neural magic to implement this and I saw you already have one reverted pr https://github.com/vllm-project/vllm/pull/7764/files#diff-db60d9d797517c6bdf2ce051b532fe6fa6bceab04516c6d8c2d80a2ae803abe9 that contains the compressed tensor format checking and loading. Do you suggest me to wait for the one to be checkin in again with marlin kernels or you have some better suggestions to help checkin this pr? I think one way for me is to check in the kernel part only in this pr. Or I can put merge together into the ExpertInt8 class to have another flag for w8a8 https://github.com/vllm-project/vllm/pull/7415/files#diff-312c4aa092501cccba5cce56647f4d812f843165ab0e2e8d1c4fd449d6ed60a8R125 . But this an extra config passing from use to know whether they wanna use w8a8 or w8a16, which is not good. I'm also not sure what specific quantization method that the expert_int8 pr is using which i'm assuming not GPTQ/AWQ as in compressed tensor format.  

I can also wait for you pr to be merged again with marlin kernels because otherwise I'll need to put  some extra checking here https://github.com/vllm-project/vllm/pull/7764/files#diff-e6cba5598f37013ace29f382dd3353f55275bd7137fb30a11e21155d73db9400L26 to only keep the case for w8a8 + compressed tensor format. Thank you!

Reply:
@mgoin I made a modification and if you think this solution is good as a temporary solution (before your more general version of code + marlin checked in), please feel free to provide comments and I can add some testing as well before checking in. Thank you!

Reply:
Sorry for missing these comments @qingquansong ! The Marlin MoE PR should be in now so I think it should be good for you to build on. For now I am willing to live with separate w8a8 and w8a16 configs in the triton kernel to get this feature landed, but we should replace this with at least an enum so we can reduce it to a single argument soon.

Reply:
Thanks for the response! Do you suggest to remove this class https://github.com/vllm-project/vllm/pull/6978/files#diff-eddafffeb6f159f8c75f635d18a502fcfbf662a562b1ae7a8683a9790161a10bR135 if the llm compressor layer loading/setup schema has been fully checked in? 🤔